backend/functions/utils.py

```diff
@@ -0,0 +1,192 @@
+                "status": status.HTTP_400_BAD_REQUEST,
+            }
+
+    except dataset_models.DatasetInstance.DoesNotExist:
+        return {
+            "message": "Output dataset instance does not exist! Create a TranslationPair DatasetInstance",
+            "status": status.HTTP_404_NOT_FOUND,
+        }
+
+    return {"message": "Success", "status": status.HTTP_200_OK}
+
+
+def get_batch_translations_using_indictrans_nmt_api(
+    sentence_list, source_language, target_language
+):
+
+    """Function to get the translation for the input sentences using the IndicTrans NMT API.
+
+    Args:
+        sentence_list (str): List of sentences to be translated.
+        source_language (str): Original language of the sentence.
+        target_language (str): Final language of the sentence.
+
+    Returns:
+        list: List of dictionaries containing the translated sentences.
+    """
+
+    # Get the translation model ID
+    model_id = LANG_TRANS_MODEL_CODES.get(
+        f"{source_language}-{target_language}", DEFAULT_ULCA_INDIC_TO_INDIC_MODEL_ID
+    )
+
+    # Convert language names to the language code
+    source_language = LANG_NAME_TO_CODE_ULCA[source_language]
+    target_language = LANG_NAME_TO_CODE_ULCA[target_language]
+
+    # Create the input sentences list
+    input_sentences = [{"source": sentence} for sentence in sentence_list]
+
+    json_data = {
+        "input": input_sentences,
+        "config": {
+            "modelId": model_id,
+            "language": {
+                "sourceLanguage": source_language,
+                "targetLanguage": target_language,
+            },
+        },
+    }
+
+    try:
+        response = requests.post(
+            "https://nmt-models.ulcacontrib.org/aai4b-nmt-inference/v0/translate",
+            json=json_data,
+        )
+
+        translations_output = response.json()["output"]
+
+        # Collect the translated sentences
+        return [translation["target"] for translation in translations_output]
+
+    except Exception as e:
+        return str(e)
+
+
+def get_translation_using_cdac_model(input_sentence, source_language, target_language):
+    """Function to get the translation for the input sentences using the CDAC model.
+
+    Args:
+        input_sentence (str): Sentence to be translated.
+        source_language (str): Original language of the sentence.
+        target_language (str): Final language of the sentence.
+
+    Returns:
+        str: Translated sentence.
+    """
+
+    # Get the translation model ID
+    model_id = LANG_TRANS_MODEL_CODES.get(
+        f"{source_language}-{target_language}", DEFAULT_ULCA_INDIC_TO_INDIC_MODEL_ID
+    )
+
+    # Convert language names to the language code
+    source_language = LANG_NAME_TO_CODE_ULCA[source_language]
+    target_language = LANG_NAME_TO_CODE_ULCA[target_language]
+
+    headers = {
+        # Already added when you pass json= but not when you pass data=
+        # 'Content-Type': 'application/json',
+    }

```

Comment:
Please remove this as well. Missed it in the earlier review sorry

Reply:
Yeah, my bad. Missed this one previously.